File: clients/rust/src/models/append_to_stream_in.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::msg_in::MsgIn;

#[non_exhaustive]
#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct AppendToStreamIn {
    pub msgs: Vec<MsgIn>,

    #[serde(rename = "streamId")]
    pub stream_id: String,
}

impl AppendToStreamIn {
    pub fn new(msgs: Vec<MsgIn>, stream_id: String) -> Self {
        Self { msgs, stream_id }
    }
}

File: clients/javascript/src/models/appendToStreamIn.ts
// this file is @generated
import {
    type MsgIn,
    MsgInSerializer,
} from './msgIn';





export interface AppendToStreamIn {
    msgs: MsgIn[];
name: string;
}

export const AppendToStreamInSerializer = {
    _fromJsonObject(object: any): AppendToStreamIn {
        return {
            msgs: object['msgs'].map((item: MsgIn) => MsgInSerializer._fromJsonObject(item)),
            name: object['name'],
            };
    },

    _toJsonObject(self: AppendToStreamIn): any {
        return {
            'msgs': self.msgs.map((item) => MsgInSerializer._toJsonObject(item)),
            'name': self.name,
            };
    }
}